src/Common/src/System/Security/Cryptography/ECDiffieHellmanOpenSsl.cs

```diff
@@ -60,29 +63,77 @@ public override int KeySize
 
                     // Set the KeySize before FreeKey so that an invalid value doesn't throw away the key
                     base.KeySize = value;
-                    _key?.Dispose();
+
+                    ThrowIfDisposed();
+                    _key.Dispose();
```

Comment:
should this method actually throw? I'm not familiar with this code path but since we dispose here and then reopen feels like it should be safe to just not dispose and not throw

Reply:
The throw is preventing the reopen; otherwise `ecdh.Dispose(); ecdh.KeySize = 384;` would effectively un-dispose the object (the next use of the key would create a new one on secp384r1 as if it had just been created)